### ISSUE TITLE
Review Privileges for Waypoint

### ISSUE THREAD

**Issue by ilrudie** [istio/istio#46592]
While working on the waypoint controller I noticed that waypoints are configured with pretty high privileges. We should review if this is required and document the reason why or lower the privileges if it's not required.

From `manifests/charts/istio-control/istio-discovery/files/waypoint.yaml`:
```yaml
        securityContext:
          privileged: true
          runAsGroup: 1337
          runAsUser: 0
          capabilities:
            add:
            - NET_ADMIN
            - NET_RAW
```

**Affected product area (please put an X in all that apply)**

[X] Ambient
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

Some orgs running a policy-as-code engine like Gatekeeper may forbid privileged pods in user namespaces via admission control. Others may audit manually and object to running ambient, or at least waypoints, based on this.

**Comment by kyessenov** [comment]
Waypoint can run as regular workloads - they don't need traffic capture (and probably don't use it AFAIR).

**Comment by ilrudie** [comment]
Thanks Kuat

**Comment by ilrudie** [comment]
Closed by #46821
